iree/tools/utils/image_util.c

```diff
@@ -0,0 +1,194 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iree/tools/utils/image_util.h"
+
+#include <math.h>
+
+#include "iree/base/internal/flags.h"
+#include "stb_image.h"
+
+iree_status_t iree_tools_utils_pixel_rescaled_to_buffer(
```

Comment:
Trace code added. It seems for the example we have (28x28x1) the looping time is comparable to `iree_hal_buffer_view_create` (400ns-1us)
![Screenshot 2021-05-10 11 57 52 AM](https://user-images.githubusercontent.com/19485875/117716515-90ef3400-b18e-11eb-83d0-7a3b12a0d562.png)

Reply:
Ah yeah, we won't get useful trace metrics from such a small image :P I'm more worried about larger images (e.g. in the dynamic shape case if the model starts with a resize down from arbitrary dimensions down to 64x64, we'd want low overhead on the buffer import + resize).

Reply:
The image_util library itself doesn't support resize, since the user may prefer one resize algorithm over the others. Besides,  how stb_image.h handles the potential alpha channel in resizing can be tricky and could create unexpected result. That's the reason why `iree_tools_utils_load_pixel_data` currently performs the dimension checks.